test/Tests/src/Data/List_Spec.enso

```diff
@@ -102,34 +140,55 @@ spec = Test.group "List" <|
             State.get Number
         sum.should_equal 6
     Test.specify "should allow reversing with .reverse" <|
-        l.reverse.head.should_equal 3
+        l.reverse.first.should_equal 3
     Test.specify "should allow dropping elements from the left with `.drop`" <|
         l.drop_start 1 . should_equal (List.Cons 2 (List.Cons 3 List.Nil))
         empty.drop_start 1 . should_equal List.Nil
     Test.specify "should allow taking elements from the left with `.take_start`" <|
         l.take_start 2 . should_equal (List.Cons 1 (List.Cons 2 List.Nil))
         empty.take_start 2 . should_equal List.Nil
-    Test.specify "should allow getting the head of the list with `.head`" <|
-        l.head . should_equal 1
-        empty.head.should_fail_with Empty_Error
     Test.specify "should allow getting the tail of the list with `.tail`" <|
         l.tail . should_equal (List.Cons 2 (List.Cons 3 List.Nil))
-        empty.tail.should_fail_with Empty_Error
+        empty.tail.should_fail_with Index_Out_Of_Bounds.Error
     Test.specify "single element list.init yields Nil" <|
         (List.Cons 1 List.Nil).init . should_equal List.Nil
     Test.specify "two element list.init yields one element" <|
         (List.Cons 1 (List.Cons 2 List.Nil)).init . should_equal (List.Cons 1 List.Nil)
     Test.specify "should allow getting the init of the list with `.init`" <|
         l.init . should_equal (List.Cons 1 (List.Cons 2 List.Nil))
-        empty.init.should_fail_with Empty_Error
+        empty.init.should_fail_with Index_Out_Of_Bounds.Error
     Test.specify "should allow getting the last element of the list with `.last`" <|
         l.last . should_equal 3
-        empty.last.should_fail_with Empty_Error
+        empty.last.should_fail_with Index_Out_Of_Bounds.Error
     Test.specify "should allow getting the head of the list with `.first`" <|
         l.first . should_equal 1
-        empty.first.should_fail_with Empty_Error
-    Test.specify "should allow getting the tail of the list with `.rest`" <|
-        l.rest . should_equal (List.Cons 2 (List.Cons 3 List.Nil))
-        empty.rest.should_fail_with Empty_Error
+        empty.first.should_fail_with Index_Out_Of_Bounds.Error
+    Test.specify "should allow getting the second item of the list with `.second`" <|
+        l.second . should_equal 2
+        empty.second.should_fail_with Index_Out_Of_Bounds.Error
```

Comment:
```suggestion
        empty.second.catch.should_equal (Index_Out_Of_Bounds.Error 1 0)
```

I'd also appreciate checking the error here, even though it is rather trivial what it will be. Ideally, we should also have a check for a 1-element list for better coverage.